### PR TITLE
Add alwaysShowTitleTextInFullscreen flag

### LINKED ISSUE #8126: Add option to show title text in titlebar in fullscreen when titleBarStyle: 'hidden-inset'
* Electron version: 1.4.10
* Operating system: macOS El Capitane

We want to show a title to our users when they are in fullscreen. However, when a window has `titleBarStyle: 'hidden-inset'` the title is not shown in fullscreen when mouse reaches the top of the screen.

We saw [this issue](https://github.com/electron/electron/issues/6715) where it is described that  it would look ugly. Is it going to look like the image provided below? Because we will be satisfied with such an appearance.

image

It would be great if Electron could have an option for enabling the title to be shown when `titleBarStyle: 'hidden-inset'` is set (something like `alwaysShowTitleTextInFullscreen`). Currently we are not happy with the behavior and even thought that it was a bug.

### PR DESCRIPTION
**Pull request by shubham2892**
Fixes #8126 
Add a flag `alwaysShowTitleTextInFullscreen` which enables to show title when `titleBarStyle` is set to `hiddenInset`. Default value is `false`. 

Screenshot with flag enabled:
image

### CODE REVIEW

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -225,6 +225,9 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
       display unless hovered over in the top left of the window. These custom
       buttons prevent issues with mouse events that occur with the standard
       window toolbar buttons. **Note:** This option is currently experimental.
+  * `alwaysShowTitleTextInFullscreen` Boolean (optional) - Shows the title in
```
Comment (kevinsawicki) [inline docs/api/browser-window.md]: What do you think about a slightly shorter name like `fullscreenWindowTitle`?

Comment (shubham2892) [inline docs/api/browser-window.md]: Yeah sounds good to me, will make the change.

**Comment by poiru** [comment]
FYI, this will be the default behavior after we switch to using the custom window buttons even for `hiddenInset`. The titlebar will also shrink to the normal size in fullscreen mode when we do that.

**Comment by kevinsawicki** [comment]
Thanks for this @shubham2892, works great 👍 🚀, pushed up one minor change to match the ivar name to the param name.

**Comment by astoilkov** [comment]
One quick question here - Why the buttons and the text are not aligned vertically in the screenshot?

**Pull request merged**